File: backend/models/dhcp.py
from pydantic import BaseModel


class StaticLeaseCreate(BaseModel):
    MACAddress: str
    IPAddress: str


class StaticLeaseUpdate(BaseModel):
    IPAddress: str | None = None
    Enable: bool | None = None


class LeaseTimeRequest(BaseModel):
    leasetime: int


class PoolCreate(BaseModel):
    name: str
    interface: str


class PoolAssignRequest(BaseModel):
    MACAddress: str

File: frontend/src/app/models/dhcp.ts
export interface StaticLeaseCreate {
  MACAddress: string;
  IPAddress: string;
}

export interface StaticLeaseUpdate {
  IPAddress?: string;
  Enable?: boolean;
}

export interface LeaseTimeRequest {
  LeaseTime: number;
}

export interface PoolCreate {
  name: string;
  interface: string;
}

export interface PoolAssignRequest {
  MACAddress: string;
}
